### ISSUE TITLE
Something can go into an infinite loop

### ISSUE THREAD

**Issue by foolswood** [concert/clapi#139]
The following digest causes the relay to use 150% CPU (if you were to remove the "changed block" bit it would go through fine):
```
-> ClientData provider (TimeStamped (Time 6303 2842395136,Trpd (TrpDigest
  { trpdNamespace = Namespace {unNamespace = engine}
  , trpdPostDefs = fromList
    [ (Tagged _kinds_amp,OpDefine {odDef = PostDefinition {postDefDoc = "Amplifier", postDefArgs = AssocList {unAssocList = [(amp,[TtInt32 (Bounds Nothing Nothing)])]}}})
    , (Tagged _types_stereo,OpDefine {odDef = PostDefinition {postDefDoc = "Loudness", postDefArgs = AssocList {unAssocList = [(gain,[TtFloat (Bounds Nothing Nothing)])]}}})]
  , trpdDefinitions = fromList
    [ (Tagged _kinds_amp_array,OpDefine {odDef = ArrayDef (ArrayDefinition {arrDefDoc = "Type array", arrPostType = Just (Tagged _kinds_amp), arrDefChildType = Tagged _kinds_amp_struct, arrDefChildEditable = ReadOnly})})
    , (Tagged _kinds_amp_struct,OpDefine {odDef = StructDef (StructDefinition {strDefDoc = "Auto generated", strDefTypes = AssocList {unAssocList = [(amp,(Tagged _kinds_amp_struct_amp,ReadOnly))]}})})
    , (Tagged _kinds_amp_struct_amp,OpDefine {odDef = TupleDef (TupleDefinition {tupDefDoc = "Amplifier", tupDefTypes = AssocList {unAssocList = [(channels,TtInt32 (Bounds Nothing Nothing))]}, tupDefInterpLimit = ILUninterpolated})})
    , (Tagged _types_stereo_array,OpDefine {odDef = ArrayDef (ArrayDefinition {arrDefDoc = "Elem array", arrPostType = Just (Tagged _types_stereo), arrDefChildType = Tagged _types_stereo_struct, arrDefChildEditable = Editable})})
    , (Tagged _types_stereo_struct,OpDefine {odDef = StructDef (StructDefinition {strDefDoc = "Should be from rage?", strDefTypes = AssocList {unAssocList = [(params,(Tagged _types_stereo_struct_params,ReadOnly)),(connections,(Tagged connections,ReadOnly))]}})})
    , (Tagged _types_stereo_struct_params,OpDefine {odDef = StructDef (StructDefinition {strDefDoc = "Auto generated", strDefTypes = AssocList {unAssocList = [(gain,(Tagged _types_stereo_struct_params_gain,Editable))]}})})
    , (Tagged _types_stereo_struct_params_gain,OpDefine {odDef = TupleDef (TupleDefinition {tupDefDoc = "Loudness", tupDefTypes = AssocList {unAssocList = [(gain,TtFloat (Bounds Nothing Nothing))]}, tupDefInterpLimit = ILLinear})})
    , (Tagged elems,OpDefine {odDef = StructDef (StructDefinition {strDefDoc = "All instances of loaded types", strDefTypes = AssocList {unAssocList = [(stereo,(Tagged _types_stereo_array,Editable))]}})})
    , (Tagged types,OpDefine {odDef = StructDef (StructDefinition {strDefDoc = "All instances of loaded kinds", strDefTypes = AssocList {unAssocList = [(amp,(Tagged _kinds_amp_array,Editable))]}})})]
  , trpdData = AssocList {unAssocList =
    [ (/elems/stereo/stamp/params/gain, TimeChange (fromList [(0,(Nothing,OpSet (Time 0 0) [WireValue 1.0] IConstant))]))

    -- CHANGED BLOCK
    , (/connections/outs/L/a, ConstChange Nothing [WireValue "/elems/stereo/stamp/connections/ins/L"])
    , (/elems/stereo/stamp/connections/ins/L/a, ConstChange Nothing [WireValue "/connections/outs/L"])
    , (/connections/outs/R/a, ConstChange Nothing [WireValue "/elems/stereo/stamp/connections/ins/R"])
    , (/elems/stereo/stamp/connections/ins/R/a, ConstChange Nothing [WireValue "/connections/outs/R"])
    , (/connections/outs/ohno/a, ConstChange Nothing [WireValue "/elems/stereo/stamp/connections/ins/ohno"])
    , (/elems/stereo/stamp/connections/ins/ohno/a, ConstChange Nothing [WireValue "/connections/outs/ohno"])

    , (/kinds/amp/loaded, ConstChange Nothing [WireValue 1])
    , (/types/amp/stereo/amp, ConstChange Nothing [WireValue 2])]}
  , trpdContOps = fromList
    [ (/connections/ins,fromList [(L,(Nothing,SoAfter Nothing)),(R,(Nothing,SoAfter (Just L)))])
    , (/connections/outs,fromList [(L,(Nothing,SoAfter Nothing)),(R,(Nothing,SoAfter (Just L)))])
    , (/elems/stereo/stamp/connections/ins,fromList [(L,(Nothing,SoAfter Nothing)),(R,(Nothing,SoAfter (Just L)))])
    , (/elems/stereo/stamp/connections/outs,fromList [(L,(Nothing,SoAfter Nothing)),(R,(Nothing,SoAfter (Just L)))])]
  , trpdErrors = Mol {unMol = fromList []}})))
```

**Comment by ch3pjw** [comment]
Has this issue been fixed by the merge of #140?

**Comment by foolswood** [comment]
It's not happening anymore, so quite possibly.
